snap/validate.go

```diff
@@ -356,6 +354,21 @@ func Validate(info *Info) error {
 	return ValidateLayoutAll(info)
 }
 
+// ValidateBase validates the base field.
+func ValidateBase(info *Info) error {
+	// validate that bases do not have base fields
+	if info.Type == TypeOS || info.Type == TypeBase {
+		if info.Base != "" && info.Base != "none" {
+			return fmt.Errorf(`cannot have "base" field on %q snap %q`, info.Type, info.InstanceName())
+		}
+	}
+
+	if info.Base == "none" && (len(info.Hooks) > 0 || len(info.Apps) > 0) {
```

Comment:
At this stage, do we know about implicitly defined hooks?

Reply:
the main caller in info.go should have found those